proxy/http2/Http2Stream.cc

```diff
@@ -509,7 +509,7 @@ Http2Stream::update_write_request(IOBufferReader *buf_reader, int64_t write_len,
         parent->connection_state.send_headers_frame(this);
 
         // See if the response is chunked.  Set up the dechunking logic if it is
-        is_done = this->response_initialize_data_handling();
+        this->response_initialize_data_handling();
```

Comment:
Actually, the is_done's between response_intiailize_data_handling and response_process_data are related.  If the response is not chunked both will always return false.  If they are chunked, they both return true if the entire chunk was represent and could be read.

It looks like my patch to open source missed the check on the return from this->response_initialize_data_handling.  There should be

if (is_done) {
  send_event = VC_EVENT_WRITE_COMPLETE;
}

I'll add that and set up a new PR.